runtime-modules/forum/src/lib.rs

```diff
@@ -503,13 +503,19 @@ decl_storage! {
         pub NextCategoryId get(next_category_id) config(): T::CategoryId;
 
         /// Map thread identifier to corresponding thread.
-        pub ThreadById get(thread_by_id) config(): map T::ThreadId => Thread<T::ForumUserId, T::CategoryId, T::Moment, T::Hash>;
+        pub ThreadById get(thread_by_id) config(): double_map T::CategoryId, blake2_256(T::ThreadId) => Thread<T::ForumUserId, T::CategoryId, T::Moment, T::Hash>;
+
+        // Map category to thread. Enables searching for threads in ThreadById without knowing to which category thread belongs to.
+        pub CategoryByThread get(category_by_thread) config(): map T::ThreadId => T::CategoryId;
```

Comment:
Why was `CategoryByThread get(category_by_thread) config(): map T::ThreadId => T::CategoryId` introduced? a thread already explicitly says what category it belongs to by `Thread::category_id`

Reply:
If the problem is that you are unable to look up a thread by only its ID after the introduction of the double map, then the proper solution is to require the user to provide a combination of the thread id and category id when trying to fully identify a thread.